enumerable_methods.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i] if block_given?
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield self[i], i if block_given?
+      i += 1
+    end
+  end
+
+  def my_select
+    result = []
+    if block_given?
+      my_each { |i| result << i if yield(i) }
+    end
+    result
+  end
+
+  def my_any?(patern = false)
+    any = false
+    my_each do |i|
+      if block_given?
+        any = true if yield(i)
+      elsif patern == true
+        any = false if (patern == i) == false
+      else
+        any = false unless i != false
+      end
+    end
+    any
+  end
```

Comment:
* [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
* [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
* [ ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
* [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern